File: JavaScript/StudyPlan/LeetCode75/Q199-BinaryTreeRightSideView.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    const nodes = [];
    function dfs(node, h) {
        if(!node) return;
        nodes[h] = node.val;
        dfs(node.left, h + 1);
        dfs(node.right, h + 1);
    }
    dfs(root, 0);
    return nodes;
};

const rightSideView = (root) => {
    const arr = [];
    const queue = [root];
    if(!root) return arr;
    
    while(queue.length) {
        let n = queue.length;
        arr.push(queue[n - 1].val);
        if(n > 0) {
            for(let i = 0; i < n; i++) {
                node = queue.shift();
                if(node.left) queue.push(node.left);
                if(node.right) queue.push(node.right);
            }
        }
    }
    return arr;
}

File: Python/Leetcode75/Medium/BinaryTreeBFS/Q199-BinaryTreeRightSideView.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        nodes = []
        queue = [root]
        while len(queue):
            level_len = len(queue)
            print(level_len)
            for i in range(level_len):
                node = queue.pop(0)
                if i == level_len - 1:
                    nodes.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

        return nodes